dwave/cloud/utils.py

```diff
@@ -241,6 +242,24 @@ def strtrunc(s, maxlen=60):
     return s[:(maxlen-3)]+'...' if len(s) > maxlen else s
 
 
+def coerce_numpy_to_python(obj):
+    """Numpy object serializer with support for basic scalar types and ndarrays."""
+
+    if isinstance(obj, numpy.integer):
```

Comment:
I have a vague recollection that `isinstance` doesn't always work as expected for NumPy dtypes. There is [np.issubdtype](https://numpy.org/doc/stable/reference/generated/numpy.issubdtype.html) that is more forgiving. Unfortunately I don't remember the specific edge case.

Reply:
`np.issubdtype` would not be a good fit here, as it would pass types, not just objects (type instances). The next line `int(obj)` would fail in that case. We have pretty comprehensive [tests](https://github.com/dwavesystems/dwave-cloud-client/blob/2d537255fefc80dc2d061d9381019abdb8f4a377/tests/test_utils.py#L181-L231) for this function (they cover all NumPy scalar types, and a few array types of interest), and `ininstance` seems to work as expected.